Update "What is KPOps" section to be more catchy (#148)
bakdata/kpops: docs/docs/user/what-is-kpops.md

# What is KPOps?

> With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can [`destroy`](/user/references/cli-commands/#deploy), [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`deploy`](/user/references/cli-commands/#deploy) a pipeline[^1] to a Kubernetes cluster!

Reviewer: ```suggestion
With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can [`deploy`](/user/references/cli-commands/#deploy), [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) a Kafka pipeline[^1] to a Kubernetes cluster!
```

Author: I had it like that initially, but I think it doesn't sound good: "...destroy a Kafka pipeline to a Kubernetes cluster!". It seems I forgot to switch out the links tho..

What do you think of it now? I reworded it a bit

[^1]:
     A pipeline can consist of just consecutive [Kafka Streams applications](/user/references/components/#streamsapp) or it could also contain [Producers](/user/references/components/#producer) and/or [Kafka Connectors](/user/references/defaults/#kafkaconnector).

## Key features

- Deploy pipelines with minimal effort and without sacrificing configurability
- Easy-to-read and write pipeline definitions
- Reset or destroy pipelines gracefully
- Reset or clean apps again with a single command
- Set defaults to share settings between pipeline steps (or whole different pipelines!)

## Example

> <!-- Replace with https://raw.githubusercontent.com/bakdata/kpops-examples/main/word-count/deployment/kpops/pipeline.yaml -->

Author: done

# What is KPOps?

> With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!

Reviewer: ```suggestion
With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a Kafka pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
```

> With a couple easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a Kafka pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
> [^1]:

Reviewer 2: ```suggestion
With a couple of easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a Kafka pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
[^1]:
```

>      A pipeline can consist of just consecutive [Kafka Streams applications](/user/references/components/#streamsapp) or it could also contain [Producers](/user/references/components/#producer) and/or [Kafka Connectors](/user/references/defaults/#kafkaconnector).

Reviewer: ```suggestion
     A Kafka pipeline can consist of just consecutive [streaming applications](/user/references/components/#streamsapp), [producers](/user/references/components/#producer) and [connectors](/user/references/defaults/#kafkaconnector).
```

Reviewer: I would prefer this wording, what do you think?

Author: I agree.

## Key features

- Deploy pipelines with minimal effort and without sacrificing configurability

> - Easy-to-read and write pipeline definitions

Reviewer 3: would use this as point 1: Easy-to-read and -write ...

Author: fixed

Reviewer: don't forget to add the leading hyphen in _-write_ as proposed by Christoph

Author: good point, done

> - Easy-to-read and -write pipeline definitions
> - Deploy pipelines with minimal effort and without sacrificing configurability
> - Reset or destroy pipelines gracefully
> - Reset or clean apps again with a single command
> - Set [defaults](/user/references/defaults) to share settings within and across pipelines

Reviewer 2: @disrupted WDYT?
```suggestion
- **Deploy Kafka apps to Kubernetes**: KPOps allows to deploy consecutive Kafka Streams applications and producers using an easy-to-read and write pipeline definition.
- **Manage Kafka Connectors**: KPOps connects with your Kafka connect cluster and applies, validates, and deletes your connector configs.
- **Configure multiple pipelines and steps**: KPOps has various abstractions that simplify configuring multiple pipelines and steps within pipelines by sharing configuration between different applications, like producers or streaming applications.
- **Handle your topics and schemas**: KPOps not only creates and deletes your topics but also registers and deletes your schemas.
- **Manage the lifecycle of your components**: KPOps not only terminates your pipeline components (i.e., Kafka Streams applications) from your Kubernetes cluster, but it cleans up the (Kafka) component-related states (i.e., removing/resetting offset of consumer groups).
```

Reviewer: @raminqaf looks great!

Author: I don't like the repeating "not only" and I am not sure i **manage the lifecycle of your components** is an accurate title for that point as it only talks about destruction.

```suggestion
- **Deploy Kafka apps to Kubernetes**: KPOps allows to deploy consecutive Kafka Streams applications and producers using an easy-to-read and write pipeline definition.
- **Manage Kafka Connectors**: KPOps connects with your Kafka connect cluster and applies, validates, and deletes your connector configs.
- **Configure multiple pipelines and steps**: KPOps has various abstractions that simplify configuring multiple pipelines and steps within pipelines by sharing configuration between different applications, like producers or streaming applications.
- **Handle your topics and schemas**: KPOps not only creates and deletes your topics but also registers and deletes your schemas.
- **Remove your components**: KPOps removes your pipeline components (i.e., Kafka Streams applications) from the Kubernetes cluster AND cleans up the (Kafka) component-related states (i.e., removing/resetting offset of consumer groups).
- **Preview your pipeline changes**: With the KPOps dry-run, you can ensure your pipeline definition is set up correctly. This helps to minimize downtime and prevent potential errors or issues that could impact your production environment.
```

Reviewer 2: I like it now even more. How about we change **Remove your components** to **Clean termination of Kafka components**

Author: Even better! :) 

@disrupted wdyt?

Reviewer: great improvements all around, go for it!

> - Set defaults to share settings between pipeline steps (or whole different pipelines!)

Reviewer 2: Maybe link the default documentation here.

> - Set defaults to share settings between pipeline steps (or whole different pipelines!)

Reviewer 2: Should we update the `Future` section of the readme with these points?

Author: Im not sure tbh, do you think they are better? If not, I could instead replace these with the points from the README.

Reviewer 2: Actually not! I think both of the sections provide good insights about KPOps keept it like that

> - Set [defaults](/user/references/defaults) to share settings between pipeline steps (or whole different pipelines!)

Reviewer 3: share settings within and across pipelines

Author: fixed

## Example

<figure markdown>
  ![atm-fraud-pipeline](/images/word-count-pipeline_streams-explorer.png)

>   <figcaption>An overview of <a href="/user/getting-started/quick-start">Word-count pipeline</a> shown in <a href="https://github.com/bakdata/streams-explorer#streams-explorer" target="_blank">Streams Explorer</a></figcaption>

Author: Are you sure? 

Word-count is the name of the pipeline, to me it makes sense to capitalize it. It is also capitalized everywhere else in the documentation.

>   <figcaption>An overview of <a href="/user/getting-started/quick-start">Word-count pipeline</a> shown in <a href="https://github.com/bakdata/streams-explorer#streams-explorer" target="_blank">Streams Explorer</a></figcaption>

Reviewer 2: Okay! If you say that it is a name and you kept it like that everywhere in the docs I get that.  So fine for my side.

# What is KPOps?

With a couple of easy commands in the shell and a [`pipeline.yaml`](#example) of under 30 lines, [KPOps](/) can not only [`deploy`](/user/references/cli-commands/#deploy) a Kafka pipeline[^1] to a Kubernetes cluster, but also [`reset`](/user/references/cli-commands/#reset), [`clean`](/user/references/cli-commands/#clean) or [`destroy`](/user/references/cli-commands/#destroy) it!
[^1]:
     A Kafka pipeline can consist of consecutive [streaming applications](/user/references/components/#streamsapp), [producers](/user/references/components/#producer), and [connectors](/user/references/defaults/#kafkaconnector).

## Key features

> - **Deploy Kafka apps to Kubernetes**: KPOps allows to deploy consecutive Kafka Streams applications and producers using an easy-to-read and write pipeline definition.

Reviewer: ```suggestion
- **Deploy Kafka apps to Kubernetes**: KPOps allows to deploy consecutive Kafka Streams applications and producers using an easy-to-read and -write pipeline definition.
```
this somehow got changed again

- **Manage Kafka Connectors**: KPOps connects with your Kafka connect cluster and applies, validates, and deletes your connector configs.

> - **Configure multiple pipelines and steps**: KPOps has various abstractions that simplify configuring multiple pipelines and steps within pipelines by sharing configuration between different applications, like producers or streaming applications.

Reviewer: ```suggestion
- **Configure multiple pipelines and steps**: KPOps has various abstractions that simplify configuring multiple pipelines and steps within pipelines by sharing common configuration between different components, such as producers or streaming applications.
```

- **Handle your topics and schemas**: KPOps not only creates and deletes your topics but also registers and deletes your schemas.

> - **Clean termination of Kafka components**: KPOps removes your pipeline components (i.e., Kafka Streams applications) from the Kubernetes cluster AND cleans up the (Kafka) component-related states (i.e., removing/resetting offset of consumer groups).

Reviewer: ```suggestion
- **Clean termination of Kafka components**: KPOps removes your pipeline components (i.e., Kafka Streams applications) from the Kubernetes cluster _and_ cleans up the component-related states (i.e., removing/resetting offset of Kafka consumer groups).
```
